README.md

```diff
@@ -145,6 +153,226 @@ readData(dataInput) {
+```
+
+Its default value is `false` (disabled).
+
+## 4. Setting Connection Timeout
+
+Connection timeout is the timeout value in milliseconds for members to accept client connection requests. The following are the example configurations.
+
+**Declarative:**
+
+```json
+{
+    "network": {
+        "connectionTimeout": 6000
+    }
+}
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.connectionTimeout = 6000;
+```
+
+Its default value is `5000` milliseconds.
+
+## 5. Setting Connection Attempt Limit
+
+While the client is trying to connect initially to one of the members in the `ClientNetworkConfig.addresses`, that member might not be available at that moment. Instead of giving up, throwing an error and stopping the client, the client will retry as many as `ClientNetworkConfig.connectionAttemptLimit` times. This is also the case when the previously established connection between the client and that member goes down. The following are example configurations.
+
+**Declarative:**
+
+```json
+{
+    "network": {
+        "connectionAttemptLimit": 5
+    }
+}
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.connectionAttemptLimit = 5;
+```
+
+Its default value is `2`.
+
+## 6. Setting Connection Attempt Period
+
+Connection timeout period is the duration in milliseconds between the connection attempts defined by `ClientNetworkConfig.connectionAttemptLimit`. The following are example configurations.
+
+**Declarative:**
+
+```json
+{
+    "network": {
+        "connectionAttemptPeriod": 5000
+    }
+}
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.networkConfig.connectionAttemptPeriod = 5000;
+```
+
+Its default value is `3000` milliseconds.
+
+## 7. Enabling Hazelcast Cloud Discovery
+
+The purpose of Hazelcast Cloud Discovery is to provide clients to use IP addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, specify a token for the `discoveryToken` field and set the `enabled` field to "true". The following are example configurations.
+
+**Declarative:**
+
+```json
+{
+ "group": {
+        "name": "hazel",
+        "password": "cast"
+    },
+
+    "network": {
+        "hazelcastCloud": {
+            "discoveryToken": "EXAMPLE_TOKEN",
+            "enabled": true
+        }
+    }
+}
+
+```
+
+**Programmatic:**
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.groupConfig.name = 'hazel';
+clientConfig.groupConfig.password = 'cast';
+
+clientConfig.networkConfig.cloudConfig.enabled = true;
+clientConfig.networkConfig.cloudConfig.discoveryToken = 'EXAMPLE_TOKEN';
+```
+
+To be able to connect to the provided IP addresses, you should use secure TLS/SSL connection between the client and members. Therefore, you should set an SSL configuration as described in the previous section.
+
```

Comment:
Let's take a note to reference to the "hazelcast.cloud" documentation when it is ready.